File: MTGSearch/src/test/java/com/dbottillo/mtgsearchfree/model/CardPropertiesTest.java
package com.dbottillo.mtgsearchfree.model;

import android.test.suitebuilder.annotation.SmallTest;

import com.dbottillo.mtgsearchfree.BaseTest;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.*;

@SmallTest
public class CardPropertiesTest extends BaseTest{

    @Test
    public void CardProperties_areCorrect(){
        assertThat(CardProperties.COLOR.getNumberFromString("White"), is(0));
        assertThat(CardProperties.COLOR.getNumberFromString("Blue"), is(1));
        assertThat(CardProperties.COLOR.getNumberFromString("Black"), is(2));
        assertThat(CardProperties.COLOR.getNumberFromString("Red"), is(3));
        assertThat(CardProperties.COLOR.getNumberFromString("Green"), is(4));

        assertThat(CardProperties.COLOR.getStringFromNumber(0), is("White"));
        assertThat(CardProperties.COLOR.getStringFromNumber(1), is("Blue"));
        assertThat(CardProperties.COLOR.getStringFromNumber(2), is("Black"));
        assertThat(CardProperties.COLOR.getStringFromNumber(3), is("Red"));
        assertThat(CardProperties.COLOR.getStringFromNumber(4), is("Green"));
    }

}

File: MTGSearch/src/test/kotlin/com/dbottillo/mtgsearchfree/model/CardPropertiesTest.kt
package com.dbottillo.mtgsearchfree.model

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat
import org.junit.Rule
import org.junit.Test
import org.mockito.junit.MockitoJUnit

class CardPropertiesTest {

    @Rule @JvmField var mockitoRule = MockitoJUnit.rule()!!

    @Test
    fun `conversion from string to color and vice versa is correct`() {
        assertThat("White".toColorInt(), `is`(0))
        assertThat("Blue".toColorInt(), `is`(1))
        assertThat("Black".toColorInt(), `is`(2))
        assertThat("Red".toColorInt(), `is`(3))
        assertThat("Green".toColorInt(), `is`(4))

        assertThat(0.toColor(), `is`("White"))
        assertThat(1.toColor(), `is`("Blue"))
        assertThat(2.toColor(), `is`("Black"))
        assertThat(3.toColor(), `is`("Red"))
        assertThat(4.toColor(), `is`("Green"))
    }

}